src/resources/auth/auth.service.ts

```diff
@@ -0,0 +1,198 @@
+import {User, UserStatus} from '../user/user.interface'
+import {generateResetToken} from '../../utils/util'
+import {Message, sendEmail} from '../../services/mail'
+import config from '../../config'
+import {newToken} from '../../utils/auth'
+import createLogger from '../../utils/logger'
+import {Token} from './auth.interface'
+import apiError, {ErrorCode} from '../../utils/apiError'
+import UserModel from '../user/user.model'
+
+const logger = createLogger(module)
+
+/**
+ * Sign up user
+ *
+ * @param newUser
+ * @param activateUserPath
+ */
+export const signup = async (
+	newUser: User,
+	activateUserPath: string,
```

Comment:
I don't think it is necessary to pass `activateUserPath` here to signup service. My point is that our service layer is where we handle all queries and operations. Thus, our controllers are getting dummier. Therefore, data like this path should be handle in here only.
Besides, concerning our philosophy that when we change something, this service layer is where we go in, I believe that `activateUserPath` should be defined here and used here

Reply:
Get your idea. But how do you construct the path? I dont want that the service layer has sth to do with req, res. The controller should process data from req, res and pass it to service

Reply:
Yeah this sounds true. Then let's go with this